src/test/java/com/adyen/ClientTest.java

```diff
@@ -38,6 +48,83 @@ public void testConfigLiveClient() {
         Assert.assertEquals(Environment.LIVE, client.getConfig().getEnvironment());
     }
 
+    private static Stream<Arguments> provideCloudTestEndpointTestCases() {
+        return Stream.of(
+            Arguments.of(null, Environment.TEST, "https://terminal-api-test.adyen.com"),
+            Arguments.of(Region.EU, Environment.TEST, "https://terminal-api-test.adyen.com"),
+            Arguments.of(Region.AU, Environment.TEST, "https://terminal-api-test.adyen.com"),
+            Arguments.of(Region.US, Environment.TEST, "https://terminal-api-test.adyen.com"),
+            Arguments.of(Region.APSE, Environment.TEST, "https://terminal-api-test.adyen.com")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideCloudTestEndpointTestCases")
+    public void testGetCloudEndpointForTestEnvironment(Region region, Environment environment, String expectedEndpoint) {
+        Config testConfig = new Config();
+        testConfig.setEnvironment(Environment.TEST);
+        Client testClient = new Client(testConfig);
+
+        // call the method and assert the result
+        String result = testClient.getCloudEndpoint(region, testConfig.getEnvironment());
+        Assert.assertEquals("Unexpected endpoint for region: " + region, expectedEndpoint, result);
+    }
+
+    private static Stream<Arguments> provideCloudLiveEndpointTestCases() {
+        return Stream.of(
+            Arguments.of(null, Environment.LIVE, "https://terminal-api-live.adyen.com"),
+            Arguments.of(Region.EU, Environment.LIVE, "https://terminal-api-live.adyen.com"),
+            Arguments.of(Region.AU, Environment.LIVE, "https://terminal-api-live-au.adyen.com"),
+            Arguments.of(Region.US, Environment.LIVE, "https://terminal-api-live-us.adyen.com"),
+            Arguments.of(Region.APSE, Environment.LIVE, "https://terminal-api-live-apse.adyen.com")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideCloudLiveEndpointTestCases")
+    public void testGetCloudEndpointForLiveEnvironment(Region region, Environment environment, String expectedEndpoint) {
+        // Create a Config object for the LIVE environment
+        Config liveConfig = new Config();
+        liveConfig.setEnvironment(Environment.LIVE);
+        Client liveClient = new Client(liveConfig);
+        
+        // call the method and assert the result
+        String result = liveClient.getCloudEndpoint(region, liveConfig.getEnvironment());
+        Assert.assertEquals("Unexpected endpoint for region: " + region, expectedEndpoint, result);
+    }
+
+    @Test
+    public void testRegionsNotInMappingThrowException() {
+        List<Region> unmappedRegions = new ArrayList<>();
```

Comment:
Since India is currently the only region that is not in the map, we can rewrite `testRegionsNotInMappingThrowException` and assert that "In"<->mapping should throw an exception, that will keep simplify this test

Reply:
I was thinking about the possibility more regions would be added without a region url in the future but it makes more sense to now only test for "In"